bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -71,11 +80,15 @@
     final long ledgerId;
     long lastAddPushed;
     volatile long lastAddConfirmed;
+    volatile long explicitLastAddConfirmed;
```

Comment:
did quite a bit code refactoring. Moved all of the explicit lac logic (write/flush path) to LacFlushPolicy/ExplicitLacFlushPolicy new class. Now code changes to LedgerHandle class is minimal.